src/python/pants/rules/core/run.py

```diff
@@ -71,7 +71,7 @@ class Run(Goal):
         console.write_stderr(f"{target} failed with code {result.process_exit_code}!\n")
 
     except Exception as e:
-      console.write_stderr(f"Exception when attempting to run {target} : {e}\n")
+      console.write_stderr(f"Exception when attempting to run {target}: {e!r}\n")
```

Comment:
Is this what we should be using elsewhere in exception strings?

Reply:
Eh, we haven't reached consensus. I think Asher and I (or maybe it was Greg) generally vote yes because the `repr` will output the exception type, like `ValueError("bad")` instead of simply `"bad"`. But, John was not convinced and thinks generally it's a bad idea because it shows possibly irrelevant exceptions to the end-user, ~leaky API.